src/main/java/org/jabref/gui/externalfiles/ImportHandler.java

```diff
@@ -159,6 +159,7 @@ public List<ImportFilesResultItemViewModel> call() {
                                     entry.clearField(StandardField.FILE);
                                     // Modifiers do not work on macOS: https://bugs.openjdk.org/browse/JDK-8264172
                                     // Similar code as org.jabref.gui.preview.PreviewPanel.PreviewPanel
+                                    generateKeys(Collections.singletonList(entry));
```

Comment:
According to the issue description, the citation key is generated at a later stage, too.

Doe snot respect `org.jabref.logic.importer.ImporterPreferences#shouldGenerateNewKeyOnImport`